pkg/chunk/aws/s3_storage_client.go

```diff
@@ -314,7 +314,7 @@ func (a *S3ObjectClient) List(ctx context.Context, prefix, delimiter string) ([]
 				}
 
 				for _, commonPrefix := range output.CommonPrefixes {
-					commonPrefixes = append(commonPrefixes, chunk.StorageCommonPrefix(commonPrefix.String()))
+					commonPrefixes = append(commonPrefixes, chunk.StorageCommonPrefix(*commonPrefix.Prefix))
```

Comment:
AWS provides an utility for that.

```suggestion
					commonPrefixes = append(commonPrefixes, chunk.StorageCommonPrefix(aws.StringValue(commonPrefix.Prefix)))
```